### DISCUSSION TITLE
which injectX prop would be best to modify content

### DISCUSSION THREAD

**Discussion by mherda** [react-native-webview/react-native-webview#2855] [Q&A]
I'm thinking of modifying a span's innerHTML before the page gets viewed by the app user. That's all I'd need to include in the inject snippet. Which injectX prop would be the most suitable to do it? I can see there are a few in the docs. Thanks

**Comment by jake920220** [comment]
Hi, @mherda .
Unfortunately, you can inject codes when your webview is fully loaded.
And you can handle DOM after DOM is loaded too.

Why don't you put a loading views on and evaluate scripts which handles span's innerHTML, then turn loading view off.

my suggestion is like

```
loading.show() // it might be react-native's loading component, or just loading component inside the webview.
${yourWebviewRef}.current.injectJavaScript("javascript:document.getElementById(#${yourSpan}).innerHTML = ${yourText}");
```
